### ISSUE TITLE
New aws provider slow aws_ssm_parameter calls when running terraform plan

### ISSUE THREAD

**Issue by Supakadoo** [terraform-providers/terraform-provider-aws#5281]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* aws_ssm_parameter

### Terraform Configuration Files

Add upwards of 100 parameter store values.
```hcl
resource "aws_ssm_parameter" "service-s3-bucket" {
  name      = "/${var.service}/${var.environment}/S3_BUCKET"
  type      = "String"
  value     = "${aws_s3_bucket.service-messages.id}"
  overwrite = true
}
```
### Expected Behavior

Better performance on newer versions of the provider

### Actual Behavior

A `terraform plan` is taking significantly longer on aws_ssm_parameter state refresh

### Steps to Reproduce

1. Set provider to:
```hcl
provider "aws" {
  region  = "${var.region}"
  version = "1.9.0"
}
```
2. Run `terraform plan` -> takes about 2 minutes to run in our case
3. Set provider to:
```hcl
provider "aws" {
  region  = "${var.region}"
  version = "1.28.0"
}
```
4. Run `terraform init`
5. Run `terraform plan` -> takes 10+ minutes now

### Important Factoids

Abnormally large amount of param store variables in our aws account(s)

**Comment by julienduchesne** [comment]
Fixed in https://github.com/terraform-providers/terraform-provider-aws/pull/5234

**Comment by bflad** [comment]
Hi everyone 👋  Sorry for the strange behavior here. It turns out that the SSM API can return results in an unexpected manner (e.g. empty DescribeParameters pages for _all_ SSM parameters even though we asked specifically for one name) depending on which type of request filtering is used.

We just merged in #5325 which changes the filtering type to only make a single API request. It should be released later today in version 1.29.0 of the AWS provider. Huge shout out to @julienduchesne who spent a lot of time doing investigative work here and coming up with (we expect) the correct solution to reduce the number of API calls.

**Comment by bflad** [comment]
This has been released in [version 1.29.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1290-july-26-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
